src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -374,5 +374,20 @@ public virtual async Task<VersionStamp> GetTopLevelChangeTextVersionAsync(Cancel
             var textAndVersion = await this.TextAndVersionSource.GetValueAsync(cancellationToken).ConfigureAwait(false);
             return textAndVersion.Version;
         }
+
+        public bool HasTextChanged(TextDocumentState oldState, bool ignoreUnchangeableDocument)
```

Comment:
Can we call this one HasTextSourceChanged or something so it's clear this isn't doing a content check? Or at least an XML doc comment?